guides/05_custom-components/05_frontend.md

```diff
@@ -0,0 +1,218 @@
+
+The `Example.svelte` file should expose the following props:
+
+```js
+    export let value: string;
+    export let type: "gallery" | "table";
+    export let selected = false;
+```
+
+* `value`: The example value that should be displayed.
+
+* `type`: It is a variable that can be either "gallery" or "table" depending on how the examples are displayed. The "gallery" form is used when the examples correspond to a single input component, while the "table" form is used when a user has multiple input components, and the examples need to populate all of them. 
+
+* `selected` You can also adjust how the examples are displayed if a user "selects" a particular example by using the selected variable.
+
+This is the `Example.svelte` file for the code `Radio` component:
+
+```javascript
+<script lang="ts">
+	export let value: string;
+	export let type: "gallery" | "table";
+	export let selected = false;
+</script>
+
+<div
+	class:table={type === "table"}
+	class:gallery={type === "gallery"}
+	class:selected
+>
+	{value}
+</div>
+
+<style>
+	.gallery {
+		padding: var(--size-1) var(--size-2);
+	}
+</style>
+```
+
+## Handling Files
+
+If your component deals with files, these files **should** be uploaded to the backend server. 
+The `@gradio/client` npm package provides the `upload_files` and `prepare_files` utility functions to help you do this.
+
+The `prepare_files` function will convert the browser's `File` datatype to gradio's internal `FileData` type.
+You should use the `FileData` data in your component to keep track of uploaded files.
+
+The `upload_files` function will upload an array of `FileData` values to the server.
+
+Here's an example of loading files from an `<input>` element when its value changes.
+
+
+```javascript
+<script lang="ts">
+
+    import type { FileData } from "@gradio/upload";
+    import {upload, prepare_files} from "@gradio/upload";
+    export let root;
+
+    async function handle_upload(file_data: FileData[]): Promise<void> {
+        await tick();
+        await upload(file_data, root);
+    }
+
+    async function loadFiles(files: FileList): Promise<void> {
+        let _files: File[] = Array.from(files);
+        if (!files.length) {
+            return;
+        }
+        if (file_count === "single") {
+            _files = [files[0]];
+        }
+        let file_data = await prepare_files(_files);
+        await handle_upload(file_data);
+    }
+
+    async function loadFilesFromUpload(e: Event): Promise<void> {
+		const target = e.target;
+
+		if (!target.files) return;
+		await loadFiles(target.files);
+	}
+</script>
+
+<input
+    type="file"
+    on:change={loadFilesFromUpload}
+    multiple={true}
+/>
+```
+
+The component exposes a prop named `root`. 
+This is passed down by the parent gradio app and it's the url that the files will be uploaded to.
+
+For WASM support, you should get the upload function from the `Context` and pass that as the third parameter of the `upload` function.
```

Comment:
@pngwn I'm certain I didn't explain this well. Can you please take a look on monday? 🙏